### DISCUSSION TITLE
New ideas / Share ideas !

### DISCUSSION THREAD

**Discussion by Natizyskunk** [Natizyskunk/vscode-sftp#68] [Ideas]
## 👋 Welcome!
We’re using this discussion as a place to communicate about differents ideas for new potential features. We hope that you:
  * Share ideas.

To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Natizyskunk** [comment]


**Reply by Natizyskunk** [reply]
- [Add FTP active mode capability](https://github.com/mscdex/node-ftp/pull/176/files)
- [Connection: Add option to specify keep-alive behaviour on connect](https://github.com/mscdex/node-ftp/pull/192/files)

**Comment by rasulsh** [comment]
I would like to thank you for developing this useful extension.
 A feature that I think is needed is the ability to create a folder on a remote FTP server.

**Reply by Natizyskunk** [reply]
@rasulsh,
Why would you want to create a file/folder on a remote that shouldn't be available on your local machine ?
Doesn't want you to also keep this in sync ?

**Comment by andremuller** [comment]
When I change the files locally checking out a branch or reverting changes/commits, does the server gets updated?

**Reply by andremuller** [reply]
I managed to make it work in these git use cases using the watcher feature.

https://github.com/Natizyskunk/vscode-sftp/wiki/Common-Config#watcher

Used the following config:

```
  "uploadOnSave": false,
  "watcher": {
        "files": **/*",
        "autoUpload": true,
        "autoDelete": true
 }
```

**Reply by Natizyskunk** [reply]
You can find the right configuration  [here](https://github.com/Natizyskunk/vscode-sftp/blob/master/FAQ.md#automatically-sync-both-ways-without-user-interaction). 🙂

**Reply by andremuller** [reply]
Thanks for the reply @Natizyskunk 😃

**Comment by mdurchholz** [comment]
I just downloaded this extension and it is great so far! I'd love to be able to disconnect from the server with a button click of some sorts. It seems as though the extension automatically connects you to the server and the only way to "disconnect" is by fudging up the sftp.json file. Is there a way to toggle the server connection?

**Reply by Natizyskunk** [reply]
@satiromarra Could you maybe take a look at it ?

**Reply by satiromarra** [reply]
Something like an icon or an option in the menu that allows “Disconnect”, It would be possible.

**Reply by Natizyskunk** [reply]
@satiromarra Would you be able to add a button with a nice icon ?

**Reply by satiromarra** [reply]
@Natizyskunk Yeah, I am working on it in 3 different developments, let's see which one is the most effective.

**Reply by Natizyskunk** [reply]
@satiromarra nice, let me know if you need help 🙂

**Reply by mdurchholz** [reply]
@satiromarra - Any updates on this feature?

**Comment by chargerback-ron** [comment]
Does the extension warn you if files on the server side have been modified since you last downloaded them? This is an important feature if more than one person is working on the project at the same time.

**Reply by Natizyskunk** [reply]
Right now, sadly no. But it's in progress.

**Comment by david-dudek** [comment]
Thank you for creating this tool.  
I have a request for something like a .gitignore file just for the sftp site.  For example, if I'm using VS Code for a website that I use the sftp tool to sync my files with, I might have files that I don't want on the webserver.

**Reply by Natizyskunk** [reply]
@david-dudek,
You can find how to do it in the WIKI.

For what you want, I recommend taking a look at the ['ignore'](https://github.com/Natizyskunk/vscode-sftp/wiki/Configuration#ignore), ['ignoreFile'](https://github.com/Natizyskunk/vscode-sftp/wiki/Configuration#ignorefile) and ['remoteExplorer.filesExclude'](https://github.com/Natizyskunk/vscode-sftp/wiki/Configuration#remoteexplorerfilesexclude) parameters.

**Comment by dickinsonk9** [comment]
Is it possible to create templates besides the basic SFTP one that we could call when we use the CMD/CTRL + Shift + P command? One specifically to create multiple profiles and another template for multiple servers? Maybe to name the templates like SFTP: Profile (I realize there is a command called SFTP: Set Profile) or SFTP: Multiple Context or SFTP: Hop(s).

It would make it easier not to constantly refer back to the extension to copy each time a new project is created. 

I use this extension everyday and love the ease of use it provides for connecting to different sites.

**Comment by sumonst21** [comment]
the  `remoteTimeOffsetInHours` is a important thing, could you make it appear by default with the initial config template like when we press CTRL+SHIFT+P?

**Reply by Natizyskunk** [reply]
@sumonst21 It should be possible yes, I'll take a look at it.
